25-multiplecomponents-lab_v16/src/01-redspotter.jsx

```diff
@@ -1,13 +1,58 @@
-/**
- * Lab 01:
- *
- * Write a game that shows 10 squares: 9 gray and 1 red.
- * When a user clicks on the red she gets 5 points and red changes its place.
- * When a user clicks on any other square nothing happens.
- *
- */
-
-import React from 'react';
-
-export default class RedSpotter extends React.Component {
-}
+import React from 'react';
+
+export default class RedSpotter extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            score: 0,
+            boxes: [
+                { id: 10, color: "#ff0000" },
+                { id: 20, color: "#c0c0c0" },
+                { id: 30, color: "#c0c0c0" },
+                { id: 40, color: "#c0c0c0" },
+                { id: 50, color: "#c0c0c0" },
+                { id: 60, color: "#c0c0c0" },
+                { id: 70, color: "#c0c0c0" },
+                { id: 80, color: "#c0c0c0" },
+                { id: 90, color: "#c0c0c0" },
+                { id: 100, color: "#c0c0c0" }
+            ]
+        };
+    }
+
+    shuffle = (id) => {
+        var _ = require('lodash');
+
+        if (id === "new") {
+            this.setState({ boxes: _.shuffle(this.state.boxes) });
+            this.setState({ score: 0 });
+        } else {
+            (id === 10 ?
+                this.setState({ score: parseInt([this.state.score]) + 10 }):
+                this.setState({ score: parseInt([this.state.score]) - 5 })
+            );
+            
+            this.setState({ boxes: _.shuffle(this.state.boxes) });
+        }
+    }
+
+    render() {
+        return (
+            <div>
+                <input type="button" value="New Game" onClick={() => this.shuffle("new")} ref={btn => { this.btn = btn }} /> Score : <label>{this.state.score}</label>
+                <p />
+                <div>
+                    {this.state.boxes.map((element, index) => {
+                        return <div key={index} id={element.id} style={{

```

Comment:
Better to define the `style` object at the top of the `render` function (and not inside the JSX)